Revive unit tests

ChrisdeWolf opened the pull request:
Code formatting:
* Ran Prettier Code formatter on the repo - no library logic was been modified just formatted

Revived Unit tests:
* Tried working on adding additional properties support but found tests were no longer compatible with most recent mongoose (https://github.com/Automattic/mongoose/blob/master/CHANGELOG.md#600-rc0--2021-08-03). Additionally, mockgoose has been deprecated: https://github.com/Mockgoose/Mockgoose. 
* In order to get tests back to workable i upgraded them to use recommended mockgoose alternative.
* Also upgraded to use async/await syntax for readability

ChrisdeWolf committed:
ran prettier code formatter on the repo

ChrisdeWolf committed:
revive unit tests to be compatible with latest mongoose and replaced deprecated mockgoose test dependency

ChrisdeWolf committed:
remove old mockgoose import comments

pibi commented:
@ChrisdeWolf, sorry for the late reply and thanks a lot for this PR. Just a few questions/comments:

1.  Does this change supports old mongoose version too?
2. I like the async/await port, but this means we are dropping support for old nodejs version. this must be stated in the package.json under the `engine` property. Can you add it?
3. this intrdoucesbreaking changes (see above), thereofre we need to bump up the version number and introduce a changelog.
4. I see a commit regarding the linting. Maybe I'm missing something, but we are linting the code before pushing it, so I'm wondering what lint you are using. BTW, can you rollback that commit?

Thanks!

ChrisdeWolf committed:
Revert "ran prettier code formatter on the repo"

This reverts commit cdc69993f37ecd1822a3e6bf39d455e58c2273d8.

ChrisdeWolf committed:
Specified minimum node version in engines property to support async/await

ChrisdeWolf committed:
updated version and added CHANGELOG

ChrisdeWolf committed:
fix unit test instance comparisons to check against ObjectID to resolve failing tests

ChrisdeWolf commented:
@pibi , no problem!

1. This change works for any mongoose version >=6.0.0 (looks like a few units break with 5.x). I couldn't find a good way to specify >=6.0.0 in package.json for mongoose without it still downloading latest versions of mongoose by default, so i added it to the CHANGELOG.
2. Specified minimum node version in `engines` property in package.json to support async/await.
3. I bumped the version number (major version) to 1.0.0 based on semantic versioning. I also added the changelog and entry.
4. I rolled this commit back. I am using default ESLint: https://eslint.org/ for my auto-linting in VSCode. What are you using and i can match?

ChrisdeWolf committed:
add entry in CHANGELOG for mongoose breaking change

ChrisdeWolf commented:
^bumping this PR @pibi
